front/components/actions/tables_query/TablesQueryActionDetails.tsx

```diff
@@ -92,78 +93,64 @@ function QueryThinking({ action }: { action: TablesQueryActionType }) {
   );
 }
 
-type TablesQueryActionWithResultsType = TablesQueryActionType & {
-  output: {
-    query_title?: string;
-    results: unknown[];
-  };
-};
-
-function hasQueryResults(
-  action: TablesQueryActionType
-): action is TablesQueryActionWithResultsType {
-  const { output } = action;
-
-  return (
-    output !== null &&
-    typeof output === "object" &&
-    "results" in output &&
-    Array.isArray(output.results)
-  );
-}
-
-function QueryTablesResults({ action }: { action: TablesQueryActionType }) {
+function QueryTablesResults({
+  action,
+  owner,
+}: {
+  action: TablesQueryActionType;
+  owner: WorkspaceType;
+}) {
   const sendNotification = useContext(SendNotificationsContext);
-
-  const handleDownload = useCallback(
-    (title: string) => {
-      if (!hasQueryResults(action)) {
-        return null;
-      }
-      if (
-        !action.output ||
-        !action.output.results ||
-        action.output.results.length === 0
-      ) {
-        return;
+  const { output } = action;
+  const title = getTablesQueryResultsFileTitle({ output });
+
+  const handleDownload = useCallback(() => {
+    if (action.resultsFileId) {
+      try {
+        const downloadUrl = `/api/w/${owner.sId}/files/${action.resultsFileId}?action=download`;
```

Comment:
It should work with `<Link>` this is how it's being done for the Citation.

Reply:
nice !